From a92a8733b1e402a0c6d76e92932872302c79fa76 Mon Sep 17 00:00:00 2001
From: Maik Otto <m.otto@phytec.de>
Date: Thu, 25 Mar 2021 14:45:20 +0100
Subject: [PATCH] dist: fapi-config: Set Keystore to /mnt/config/tpm/tss

Upstream-Status: Pending

Signed-off-by: Maik Otto <m.otto@phytec.de>
Signed-off-by: Yannic Moog <y.moog@phytec.de>
---
 dist/fapi-config.json.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dist/fapi-config.json.in b/dist/fapi-config.json.in
index f2388f21..3967ad64 100644
--- a/dist/fapi-config.json.in
+++ b/dist/fapi-config.json.in
@@ -1,8 +1,8 @@
 {
      "profile_name": "P_ECCP256SHA256",
      "profile_dir": "@sysconfdir@/tpm2-tss/fapi-profiles/",
-     "user_dir": "~/@userstatedir@/tpm2-tss/user/keystore",
-     "system_dir": "@localstatedir@/lib/tpm2-tss/system/keystore",
+     "user_dir": "/mnt/config/tpm2/tss/user/keystore",
+     "system_dir": "/mnt/config/tpm2/tss/system/keystore",
      "tcti": "",
      "system_pcrs" : [],
      "log_dir" : "@runstatedir@/tpm2-tss/eventlog/",
--
2.34.1

