From b454d318194d61e806fdbbf0d3c84deae6e07861 Mon Sep 17 00:00:00 2001
From: Wadim Egorov <w.egorov@phytec.de>
Date: Fri, 18 Nov 2016 10:21:53 +0100
Subject: [PATCH] Add rockchip to modules

Signed-off-by: Wadim Egorov <w.egorov@phytec.de>
---
 kmscube.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kmscube.c b/kmscube.c
index ca2c266..6a72f11 100644
--- a/kmscube.c
+++ b/kmscube.c
@@ -130,6 +130,7 @@ static int init_drm(void)
 		"vc4",
 		"virtio_gpu",
 		"vmwgfx",
+		"rockchip",
 	};
 	drmModeRes *resources;
 	drmModeConnector *connector = NULL;
-- 
1.9.1

