From 4a170ca6ba06f2954f5d4c692448e39a2833d12b Mon Sep 17 00:00:00 2001
From: Wadim Egorov <w.egorov@phytec.de>
Date: Fri, 18 Nov 2016 10:04:44 +0100
Subject: [PATCH 1/2] Use GBM platform specific functions

Signed-off-by: Wadim Egorov <w.egorov@phytec.de>
---
 kmscube.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/kmscube.c b/kmscube.c
index 2356b7f..c497490 100644
--- a/kmscube.c
+++ b/kmscube.c
@@ -41,6 +41,7 @@
 
 #define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]))
 
+static PFNEGLGETPLATFORMDISPLAYEXTPROC eglGetPlatformDisplayEXT = NULL;
 
 static struct {
 	EGLDisplay display;
@@ -331,7 +332,12 @@ static int init_gl(void)
 			"    gl_FragColor = vVaryingColor;  \n"
 			"}                                  \n";
 
-	gl.display = eglGetDisplay(gbm.dev);
+	eglGetPlatformDisplayEXT = (void *) eglGetProcAddress("eglGetPlatformDisplayEXT");
+	if (!eglGetPlatformDisplayEXT) {
+		printf("could not get address of eglGetPlatformDisplayEXT\n");
+		return -1;
+	}
+	gl.display = eglGetPlatformDisplayEXT(EGL_PLATFORM_GBM_KHR, gbm.dev, NULL);
 
 	if (!eglInitialize(gl.display, &major, &minor)) {
 		printf("failed to initialize\n");
-- 
1.9.1

