From 6944d9acfef05b14471b5deb153fcf0bed89f0ea Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20M=C3=BCller-Klieser?= <s.mueller-klieser@phytec.de>
Date: Thu, 26 Jan 2017 00:02:28 +0100
Subject: [PATCH] project: fix install path for poky
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-status: not applicable, poky specific

Signed-off-by: Stefan Müller-Klieser <s.mueller-klieser@phytec.de>
---
 src/src.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/src.pro b/src/src.pro
index 79e4bdc..f5c9269 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -38,6 +38,6 @@ RESOURCES += resources.qrc
     HEADERS += touchmockingapplication.h
     QT += gui-private
 } else {
-    target.path =/data/user/qt/qtwebbrowser-app
+    target.path = $$(bindir)
     INSTALLS += target
 }
-- 
1.9.1

