From 4f564a5ec2467af4a035acf6b123baaeb679dbe9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20M=C3=BCller-Klieser?= <s.mueller-klieser@phytec.de>
Date: Tue, 11 Jul 2017 14:01:18 +0200
Subject: [PATCH] main.qml: remove imx6ul non working demos
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stefan Müller-Klieser <s.mueller-klieser@phytec.de>
---
 main.qml | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/main.qml b/main.qml
index c70f4c8..cc841c8 100644
--- a/main.qml
+++ b/main.qml
@@ -51,12 +51,6 @@ Window {
     ListModel {
         id: menue_model
         ListElement {
-            name: "Multimedia"
-            icon: "images/icon_multimedia.png"
-            pageSource: "content/PageMultimedia.qml"
-            description: "PLAY VIDEOS"
-        }
-        ListElement {
             name: "Image<br>Viewer"
             icon: "images/icon_image_viewer_128x128.png"
             pageSource: "content/PageImageViewer.qml"
@@ -69,18 +63,6 @@ Window {
             description: "VISUALIZE MEASURED DATA"
         }
         ListElement {
-            name: "Multitouch<br>Demo"
-            icon: "images/icon_multitouch.png"
-            pageSource: "content/PageMultitouch.qml"
-            description: "USE MULTITOUCH"
-        }
-        ListElement {
-            name: "WebBrowser"
-            icon: "images/icon_web_128x128.png"
-            pageSource: "content/PageWebbrowser.qml"
-            description: "BROWSE THE WEB"
-        }
-        ListElement {
             name: "Virtual Keyboard Test"
             icon: "qrc:/images/icon_vkey_128x128.png"
             pageSource: "qrc:/content/PageVirtualKeyboardTest.qml"
-- 
1.9.1

